src/Symfony/Component/String/LazyString.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+/**
+ * A string whose value is computed lazily by a callback.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class LazyString implements \JsonSerializable
+{
+    private $value;
+
+    /**
+     * @param callable $callback A callable or a [Closure, method] lazy-callable
+     *
+     * @return static
+     */
+    public static function fromCallable($callback, ...$arguments): self
```

Comment:
Why don't you typehint `callable`?

Reply:
because we also accept `[Closure, method]` lazy-callables

Reply:
Why do you want to support that? End user can wrap `[Closure, method]` in another `Closure`

Reply:
Because we do support them e.g. in EventDispatcher and they're useful.